packages/navigation/src/UIGridList/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { ListRenderItem, View, StyleSheet } from 'react-native';
+import { FlatList } from '..';
+import { UIConstant } from '../constants';
+import type { GridProps } from './types';
+
+export function UIGridList<T>({
+    data,
+    renderItem: renderItemProp,
+    containerStyle,
+    itemHeight,
+    testID,
+}: GridProps<T>) {
+    const renderItem: ListRenderItem<T> = React.useCallback(
+        ({ item, index, separators }) => {
+            const height = itemHeight ? { height: itemHeight } : styles.itemSquare;
+            return (
+                <View style={[styles.item, height]}>
+                    {renderItemProp({ item, index, separators })}
+                </View>
+            );
+        },
+        [renderItemProp, itemHeight],
+    );
+
+    return (
+        <FlatList
```

Comment:
I also thought, that since we know height, then we can apply optimisation by default! I'm talking about https://reactnative.dev/docs/0.64/flatlist#getitemlayout

Reply:
@savelichalex but we do not know the height if it is not passed by itemHeight. By default used aspectRatio: 1 and flex: 0.5. You would like to calculate the height if passed itemHeight?

Reply:
Maybe apply it only if there was height provided? 🤔